### ISSUE TITLE
Build failure under Xcode 8.2

### ISSUE THREAD

**Issue by geoo993** [typelift/Swiftz#320]
Version
========

13066bc6e238b1e6c567ca51ba5bda7e26104b47

Environment
===========

 macOS Sierra 10.12.2 (16C67)

Description
============

Swiftz failed to build under Mac OSX sierra and Xcode 8.2.

Steps To Reproduce
===================

Build under Xcode 8.2

Expected Result
================

Build should succeed.

Actual Result
==============

Build Failure

Swiftz/Sources/List.swift:443:10: error: ambiguous use of 'init'
                return List((initial, self.iterate(f, initial: f(initial))))
                       ^
Swiftz/Sources/List.swift:46:2: note: found this candidate
        init(_ next : @autoclosure @escaping () -> (head : Element, tail : List<Element>), isEmpty : Bool = false) {
        ^
Swiftz/Sources/List.swift:60:9: note: found this candidate
        public init(_ head : Element, _ tail : List<Element> = List<Element>()) {

Additional information
======================

_Please provide any additional information that helps debugging, e.g. log output, stack traces, error codes, etc._

**Comment by CodaFi** [comment]
This is a consequence of SE-110 being partially implemented.  

```swift
public static func iterate(_ f : @escaping (Element) -> Element, initial : Element) -> List<Element> {
  return List(initial, self.iterate(f, initial: f(initial)))
}
```

**Comment by CodaFi** [comment]
Resolved by the merge of #321.  Thanks!
